tools/pipelines-tasks/AVDAppAttachPublish/task.json

```diff
@@ -0,0 +1,142 @@
+          "helpMarkDown": "Select the Azure File Share for App Attach"
+      },
+      {
+          "name": "HostPool",
+          "type": "pickList",
+          "label": "Host Pool",
+          "defaultValue": "",
+          "required": true,
+          "properties": {
+              "EditableOptions": "True"
+          },
+          "helpMarkDown": "Select the Azure Host Pool for App Attach"
+      },
+      {
+          "name": "WorkSpace",
+          "type": "pickList",
+          "label": "Workspace",
+          "defaultValue": "",
+          "required": true,
+          "properties": {
+              "EditableOptions": "True"
+          },
+          "helpMarkDown": "Select the Azure Work Space for App Attach"
+      },
+      {
+          "name": "ApplicationGroup",
+          "type": "pickList",
+          "label": "Application Group",
+          "defaultValue": "",
+          "required": true,
+          "properties": {
+              "EditableOptions": "True"
+          },
+          "helpMarkDown": "Select the Azure Application Group for App Attach"
+      }
+	],
+	"dataSourceBindings": [
+		{
+			"target": "ResourceGroupName",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"dataSourceName": "AzureResourceGroups"
+		},
+		{
+			"target": "StorageAccount",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"endpointUrl": "{{endpoint.url}}subscriptions/{{endpoint.subscriptionId}}/resourcegroups/$(ResourceGroupName)/providers/Microsoft.Storage/storageAccounts?api-version=2023-01-01",
+			"resultSelector": "jsonpath:$.value[*].name"
+		},
+		{
+			"target": "FileShare",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"endpointUrl": "{{endpoint.url}}subscriptions/{{endpoint.subscriptionId}}/resourcegroups/$(ResourceGroupName)/providers/Microsoft.Storage/storageAccounts/$(StorageAccount)/fileServices/default/shares?api-version=2023-01-01",
+			"resultSelector": "jsonpath:$.value[*].name"
+		},
+		{
+			"target": "HostPool",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"endpointUrl": "{{endpoint.url}}subscriptions/{{endpoint.subscriptionId}}/resourcegroups/$(ResourceGroupName)/providers/Microsoft.DesktopVirtualization/hostPools?api-version=2022-02-10-preview",
+			"resultSelector": "jsonpath:$.value[*].name"
+		},
+		{
+			"target": "WorkSpace",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"endpointUrl": "{{endpoint.url}}subscriptions/{{endpoint.subscriptionId}}/resourcegroups/$(ResourceGroupName)/providers/Microsoft.DesktopVirtualization/workspaces?api-version=2022-02-10-preview",
+			"resultSelector": "jsonpath:$.value[*].name"
+		},
+		{
+			"target": "ApplicationGroup",
+			"endpointId": "$(ConnectedServiceNameARM)",
+			"endpointUrl": "{{endpoint.url}}subscriptions/{{endpoint.subscriptionId}}/resourcegroups/$(ResourceGroupName)/providers/Microsoft.DesktopVirtualization/applicationGroups?api-version=2022-02-10-preview",
+			"resultSelector": "jsonpath:$.value[*].name"
+		}
+	],
+    "execution": {
+		"Node10": {
```

Comment:
This would make to install two node versions for the project, Node 10 for other tasks and Node 16 for AVD Publish. Also, Node10 is required for tests as higher version created problems. Hence, remaining it as Node10.